src/app/components/pages/Welcome.jsx

```diff
@@ -8,11 +8,11 @@ class Welcome extends React.Component {
                 <div className="column large-8 medium-10 small-12">
                     <div className="Welcome__banner">
                         <div className="Welcome__welcome">Welcome to</div>
-                        <img src={require(`app/assets/images/welcome.jpg`)} />
+                        <img src={require('app/assets/images/welcome.jpg')} />
```

Comment:
the name `require` for anything other than the commonjs module importer confuses me. could we think of a better name for this helper?

Reply:
This is webpack's magic :)
To make it work server side we use webpack-isomorphic-tools

Reply:
Sorry, I'm confused. Can you point me to the definition of this `require` function and explain the magic a bit? It seems to me as though we just need a path to where the image is being hosted, right? Or does this inject like a Base64-encoded version of the image inline in the HTML?

I'm pretty familiar with how JS building and bundling works, I've contributed to both AMD and SystemJS.  I've never seen a `require` used from within a JSX template.

Reply:
It can do both - inline and hosted images.
Webpack loads resources via the loaders, take a look at webpack/base.config.js:58 - it defines the rule for images: {test: /\.(jpe?g|png)/, use: 'url-loader?limit=4096'}, so if image size is less than 4kb it will use inline version if larger it will add src with corresponding path. Images embedded with links are added to server's static assets.